src/DotNetWorker.Core/FunctionsApplication.cs

```diff
@@ -64,5 +90,26 @@ public Task InvokeFunctionAsync(FunctionContext context)
                 return _functionExecutionDelegate(context);
             }
         }
+
+        public void CancelInvocation(string invocationId)
+        {
+            var invocationDetails = _functionInvocationDictionary.TryGetInvocationDetails(invocationId);
+            if (invocationDetails?.CancellationTokenSource is not null)
```

Comment:
Just to avoid any potential race condition, consider assigning via the pattern matching:

``` CSharp
if (invocationDetails?.CancellationTokenSource is CancellationTokenSource cts)
{
    cts.Cancel(); // with try/catch
}
```

The race condition here is _technically_ the `invocationDetails.CancellationTokenSource` could become `null` between your null check and accessing (because it is a property access, you don't know if the implementation can switch to null between calls). Using pattern matching assignment will capture that reference to the `cts` variable.